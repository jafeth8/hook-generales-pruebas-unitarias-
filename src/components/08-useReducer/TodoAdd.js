import React from 'react'
import { UseForm } from '../../hooks/UseForm';

export const TodoAdd = ({handleAddTodo}) => {
    const [{description}, handleChange, reset]=UseForm({description:''});

    const handleSubmit=(e)=> {
        e.preventDefault();

        if(description.trim().length<3){
            return;
        }
        
        const newTodo = {
            id:new Date().getTime(),
            desc:description,
            done: false
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <>
            <h4>Agregar TODO</h4>
            <hr/>

            <form onSubmit={handleSubmit}>
                <input type="text" name="description" 
                placeholder="Aprender..." autoComplete="off" 
                className="form-control" value={description}
                onChange={handleChange}
                />

                <button className="btn btn-outline-primary mt-1 w-100" type="submit">
                    Agregar
                </button>
            </form>     
        </>
    )
}
