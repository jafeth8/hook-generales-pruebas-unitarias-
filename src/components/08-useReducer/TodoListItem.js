import React from 'react'
import PropTypes from 'prop-types'


export const TodoListItem = ({todo,index,handleToggle,handleDelete}) => {
    return (
        <li className="list-group-item">
            <p onClick={()=>handleToggle(todo.id)} className={ todo.done? "complete" : "" } >{index+1}. {todo.desc}</p>
            <button className="btn btn-danger" onClick={()=>handleDelete(todo.id)}>borrar</button>
        </li>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
}


