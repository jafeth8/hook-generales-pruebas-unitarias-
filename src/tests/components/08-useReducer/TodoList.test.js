import {shallow} from 'enzyme'
import { TodoList } from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';



describe('pruebas en el componente <TodoList/>', () => {

    const handleDelete=jest.fn();
    const handleToggle=jest.fn();

    const wrapper = shallow(<TodoList 
        todos={demoTodos} 
        handleDelete={handleDelete}
        handleToggle={handleToggle} 
    />);

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener el numero correcto de <todoListItem/> y evaluar que unas de sus propiedades sea una funcion', () => {
       
        //console.log(wrapper.find('TodoListItem').length)
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
         //console.log(wrapper.find('TodoListItem').at(0).props());
         //console.log(wrapper.find('TodoListItem').at(0).prop('handleDelete'));
        
         expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    })
    
    
})
