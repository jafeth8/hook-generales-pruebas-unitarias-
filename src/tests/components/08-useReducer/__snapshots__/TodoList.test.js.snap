// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <TodoList/> debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="123"
    todo={
      Object {
        "desc": "Aprender laravel",
        "done": false,
        "id": 123,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="1234"
    todo={
      Object {
        "desc": "Aprender ingles",
        "done": false,
        "id": 1234,
      }
    }
  />
</ul>
`;
